README.md

```diff
@@ -1,35 +1,122 @@
-[![Build Status](https://travis-ci.org/usertesting/racym.png)](https://travis-ci.org/usertesting/racym)
+# racym
 
-[![Code Climate](https://codeclimate.com/github/usertesting/racym.png)](https://codeclimate.com/github/usertesting/racym)
-
-# Racym
+[![Travis](https://img.shields.io/travis/usertesting/racym?style=for-the-badge)](https://travis-ci.org/usertesting/racym) [![Coveralls github](https://img.shields.io/coveralls/github/usertesting/racym?style=for-the-badge)](https://coveralls.io/github/usertesting/racym) [![Code Climate maintainability](https://img.shields.io/codeclimate/maintainability/usertesting/racym?style=for-the-badge)](https://codeclimate.com/github/usertesting/racym)
 
 Rails Application Configuration for Yield Main (RACYM)
 
-Racym is used as a shortcut to rails configuration.
+racym is used as a shortcut to rails configuration.
 
 ## Installation
 
 Add this line to your application's Gemfile:
 
-    gem 'racym'
+```ruby
+gem "racym"
+```
 
 And then execute:
 
-    $ bundle
+```ruby
+$ bundle install
+```
+
+## Usage
 
-Or install it yourself as:
+### Setting config values
 
-    $ gem install racym
+In your `config/application.rb` add config settings:
 
-## Usage
+```ruby
+module MyApp
+  class Application < Rails::Application
+
+  ...
+
+  # Add a custom config setting here:
+  config.foo = "bar"
+
+  # Or a proc:
+  config.biz = -> { DateTime.current }
+```
+
+### Getting config values
+
+From anywhere in your Rails application (controllers/helpers/views - should be available anywhere!)
+
+```ruby
+puts racym(:foo)
+# => "bar"
+
+puts racym(:biz)
+# => Wed, 02 Oct 2019 11:13:20 -0700
+
+# ten minutes later:
+puts racym(:biz)
+# => Wed, 02 Oct 2019 11:23:20 -0700
+```
+
+### Multilevel key/values
+
+#### Setting:
+
+```ruby
+module MyApp
+  class Application < Rails::Application
+
+  ...
+
+  config.foo.username = "BAR"
+  config.foo.password = "BIZ"
+```
+
+#### Getting:
+
+```ruby
+puts racym(:foo, :username)
+# => "BAR"
+
+puts racym(:foo, :password)
+# => "BIZ"
+```
+
+### Testing
+
+There's some convenient methods you can use to test code using racym:
+  - `racym_set`
+  - `racym_undo!`
+
+For example, to test this simple method:
+
+```ruby
+class Foo
+  def self.hi
+    "Hi #{racym :person_to_hi}"
+  end
+end
+```
+
+```ruby
+describe Foo do
+  describe ".hi" do
+    before  { racym_set :person_to_hi, "Dave" }
+    specify { expect(Foo.hi).to eq "Hi Dave"  }
+    after   { racym_undo! }
+  end
+end
+```
+
+## License
+
+[MIT](LICENSE).
+
+Library created by [UserTesting](https://usertesting.com)
```

Comment:
Another place we could use a developers landing page.
![developers](https://media.giphy.com/media/l3q2zbskZp2j8wniE/giphy-downsized-large.gif)